File: src/main/kotlin/br/com/poc/ExemploLista.java
package br.com.poc;

import java.util.List;

public class ExemploLista {
    public static void main(String[] args) {
        // Criando uma lista em Java
        List<Integer> numerosJava = List.of(1, 2, 3, 4, 5);

        // Imprimindo os elementos da lista
        System.out.println("Elementos da lista em Java:");
        numerosJava.forEach(System.out::println);

        // Filtrando apenas números pares usando lambda
        List<Integer> numerosParesJava = numerosJava.stream()
                .filter(numero -> numero % 2 == 0)
                .toList();

        // Imprimindo os números pares
        System.out.println("Números pares em Java:");
        System.out.println(numerosParesJava);

        // Somando todos os elementos da lista usando lambda
        int somaJava = numerosJava.stream().mapToInt(Integer::intValue).sum();
        System.out.println("Soma dos números em Java: " + somaJava);

        // Calculando o produto dos elementos da lista usando reduce
        int produtoJava = numerosJava.stream().reduce(1, (acc, numero) -> acc * numero);

        // Imprimindo o produto
        System.out.println("Produto dos números em Java: " + produtoJava);

    }
}

File: src/main/kotlin/br/com/poc/ExemploLista.kt
package br.com.poc

fun main() {
    // Criando uma lista em Kotlin
    val numerosKotlin = listOf(1, 2, 3, 4, 5)

    // Imprimindo os elementos da lista
    println("Elementos da lista em Kotlin:")
    numerosKotlin.forEach(System.out::println)

//    for (numero in numerosKotlin) {
//        println(numero)
//    }

    // Filtrando apenas números pares
    val numerosParesKotlin = numerosKotlin.filter { it % 2 == 0 }

    // Imprimindo os números pares
    println("Números pares em Kotlin:")
    println(numerosParesKotlin)

    // Somando todos os elementos da lista
    val somaKotlin = numerosKotlin.sum()
    println("Soma dos números em Kotlin: $somaKotlin")

    // Calculando o produto dos elementos da lista usando reduce
    val produto = numerosKotlin.reduce { acc, numero -> acc * numero }
    // Imprimindo o produto
    println("Produto dos números em Kotlin: $produto")

}
